### PR TITLE
Scaffolding support minimal projects in .NET 6 Preview 7+

### PR DESCRIPTION
**Pull request by deepchoudhery**
Enabling scaffolding support for minimal projects in .NET 6 Preview 7. 
Following scaffolders are working with this new update :
- Area
- Razor Page
- Razor Page w/ EF
- MVC views and controllers 
- MVC views and controllers w/ EF
- API controllers
- API controllers w/ EF

Scaffolders not working yet : 
- Identity --> Tracking here.

Since there is no Startup.cs, all code changes are made to Program.cs regarding app setup code for DbContexts and any other app service calls. The scaffolder content of the controllers, views, and pages has not been changed.

Added a new shared assembly between aspnet-codegenerator and msidentity tools called "Microsoft.DotNet.Scaffolding.Shared" causing the high number of file changes (mostly namespace imports). Will require corresponding changes in VS to refer to this assembly instead of Microsoft.VisualStudio.Web.CodeGeneration.Contracts (doesn't exist anymore)

### CODE REVIEW

File: scripts/install-aspnet-codegenerator.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ rm -rf artifacts
 dotnet tool uninstall -g dotnet-aspnet-codegenerator 
 cd $DEFAULT_NUPKG_PATH
 rm -rf microsoft.visualstudio.web.codegeneration
-rm -rf microsoft.visualstudio.web.codegeneration.contracts
```
Comment (phenning) [inline scripts/install-aspnet-codegenerator.sh:16]: Does casing matter on these? For non windows builds? It’s lowercase in the previous file.

File: src/Scaffolding/VS.Web.CG.EFCore/DbContextEditorServices.cs [hunk 2]
```diff
@@ -161,39 +169,25 @@ public EditSyntaxTreeResult EditStartupForNewContext(ModelType startUp, string d
...
+                //if using Startup.cs, the ConfigureServices method should exist. 
                 if (configServicesMethod != null && configRootProperty != null)
                 {
                     var servicesParam = configServicesMethod.ParameterList.Parameters
-                        .FirstOrDefault(p => p.Type.ToString() == "IServiceCollection") as ParameterSyntax;
-                    string textToAddAtEnd;
+                        .FirstOrDefault(p => p.Type.ToString().Equals(IServiceCollection));
+                        
                     var statementLeadingTrivia = configServicesMethod.Body.OpenBraceToken.LeadingTrivia.ToString() + "    ";
+                    string textToAddAtEnd = AddDbContextString(minimalHostingTemplate: false, useSqlite, statementLeadingTrivia);
```
Comment (phenning) [inline src/Scaffolding/VS.Web.CG.EFCore/DbContextEditorServices.cs:181]: This can move inside the if block below, no?

**Review by phenning — APPROVED** [review]

File: scripts/install-aspnet-codegenerator.sh (hunk 1 above)
Comment (deepchoudhery) [reply scripts/install-aspnet-codegenerator.sh:16]: i haven't tested these on mac/linux in a while bit I remember them working. Again, these are just helper scripts that I maintain. Definitely will circle back to these.

File: src/Scaffolding/VS.Web.CG.EFCore/DbContextEditorServices.cs (hunk 2 above)
Comment (deepchoudhery) [reply src/Scaffolding/VS.Web.CG.EFCore/DbContextEditorServices.cs:181]: yep will move it.

**Pull request merged**